lib/smart_proxy_ansible/playbooks_reader.rb

```diff
@@ -0,0 +1,43 @@
+
+module Proxy
+  module Ansible
+
+    class PlaybooksReader
+      class << self
+
+        def playbooks_names
+          ReaderHelper.collections_paths.split(':').map { |path| collection_playbooks_names(path) }.flatten
+        end
+
+        def playbooks
+          ReaderHelper.collections_paths.split(':').reduce([]) { |playbooks, path | playbooks.push(read_collection_playbooks(path)) }.flatten
+        end
+
+        def read_collection_playbooks(collections_path)
+          Dir.glob("#{collections_path}/ansible_collections/*/*/playbooks/*").map do |path|
+            {
+                name: ReaderHelper.playbook_or_role_full_name(path),
+                playbooks_content: File.readlines(path)
+            }
+          end
+        rescue Errno::ENOENT, Errno::EACCES => e
+          logger.debug(e.backtrace)
+          message = "Could not read Ansible playbooks #{collections_path} - #{e.message}"
+          raise ReadPlaybooksException.new(message), message
+        end
+
+        private
+
+        def collection_playbooks_names(collections_path)
```

Comment:
It seems like `read_collection_playbooks` and `collection_playbooks_names` have a lot of common code except for the part that is `map`ped to `path`. Do you think it would be worth extracting the common parts?